src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.dsde.rawls.model
+
+import org.broadinstitute.dsde.rawls.model.WorkspaceAccessLevel._
+
+sealed trait UserAuthRef
+case class RawlsUserRef(userEmail: String) extends UserAuthRef
+case class RawlsGroupRef(groupName: String) extends UserAuthRef
+
+case class RawlsUser(userEmail: String) extends DomainObject {
+  def idFields = Seq("userEmail")
+}
+
+object RawlsUser {
+  implicit def toRef(u: RawlsUser) = RawlsUserRef(u.userEmail)
+}
+
+case class RawlsGroup(groupName: String, users: Set[RawlsUserRef], subGroups: Set[RawlsGroupRef]) extends DomainObject {
+  def idFields = Seq("groupName")
+}
+
+object RawlsGroup {
+  implicit def toRef(g: RawlsGroup) = RawlsGroupRef(g.groupName)
+}
+
+
+object UserAuth {
+
+  def toWorkspaceAccessGroupName(workspaceName: WorkspaceName, accessLevel: WorkspaceAccessLevel) =
+    s"rawls ${workspaceName.namespace}/${workspaceName.name} ${accessLevel}"
+
+}
+
+object UserAuthJsonSupport extends JsonSupport {
+  // need "apply" here so it doesn't choose the companion class
```

Comment:
No.  `jsonFormat1(RawlsUser)` on the following line does not work because the compiler matches on the `object` instead of the `case class`, causing this to fail.  This is to note the workaround of using `RawlsUser.apply` instead.